src/Bicep.LangServer.IntegrationTests/CompletionTests.cs

```diff
@@ -493,7 +493,6 @@ public async Task VerifyResourceBodyCompletionWithDiscriminatedObjectTypeContain
 		azCliVersion: $3
 		retentionInterval: $4
 	}
```

Comment:
>} [](http://example.com/codeflow?start=1&length=1)

Since the $0 is removed, where does the cursor land after the snippet is committed?

Reply:
Good question. Right now it places it outside the top level close curly brace.  Placing it at column 0 at ln 9 in the below screenshot might be simple. I am not sure if we want ln 8. That might be tricky. Let me know what you think.

![image](https://user-images.githubusercontent.com/30270536/126716950-b03246ac-6e37-4b9e-beec-c160d338e8f1.png)

Reply:
Actually, final tab stop was always before the last '}'. So added it back without the newline as part of:
https://github.com/Azure/bicep/commit/ee8fa5424566ab2a1a2fa510cad4bbada676bf63

Let me know if you have any questions/concerns.

Reply:
I think the answer depends on what the most likely action is after tab stopping through the snippet. If most of the time, the user doesn't add more properties, then we could actually put $0 after the last curly.

If we we think the user will most often add another property, then we might have to put $0 at the end of the value of the last property we inserted because pressing enter would let them add the property easily.

Reply:
Makes sense.  Should we wait for more customer feedback regarding placement of final tab stop($0)? @MarcusFelling, @alex-frankel do you have any inputs?

Reply:
I thought about it some more. My guess is the following:
* Adding more top-level properties to resources is rare, so it's probably ok to assume that the user isn't going to do it.
* There are only 1-2 top-level writable properties in modules (`name` and `params`), so users will not be adding more.
* Many swaggers don't mark all the inner (not top-level) properties as required, so I'm guessing users are more likely to add properties inside the resource.
* I don't know how likely the users are to provide values for optional module params.

Thoughts? This could get pretty complex, so I'm definitely open to keeping the logic simple. Just wanted to discuss first.

Reply:
> I think the answer depends on what the most likely action is after tab stopping through the snippet. If most of the time, the user doesn't add more properties, then we could actually put $0 after the last curly.

I agree with this solution - it feels impossible to really predict what the user's next action is going to be, so putting a `$0` after the last curly feels like the most agnostic option.

Inserting a `$0` into the 'top-level' object (the current behavior) feels like we're attempting to make a prediction that inserting another top-level property is important - but who's to say the user cares about adding top-level properties, or perhaps the property they actually want to insert isn't actually a top-level property.

Reply:
Had an offline discussion with @anthony-c-martin. Does anyone object to having the final tab stop to be outside the resource after last } at ln9 in above screenshot?

Reply:
Sounds good to me.